File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int snowball = 0;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i]==0)
                snowball++;
            else{
                swap(nums[i], nums[i-snowball]);
            }
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int snowball = 0;
        int n = nums.length;
        if(n<=1)
            return;
        for(int i = 0; i<n; i++){
            if(nums[i]==0){
                snowball++;
            }else{
                int temp = nums[i];
                nums[i] = nums[i-snowball];
                nums[i-snowball] = temp;
                
            }
        }
    }
}